go/error-handling/error_handling.go

```diff
@@ -0,0 +1,34 @@
+//This package provides functions
+package erratum
+
+// Use takes a ResourceOpener function as a parameter and attempts to
+// call it to obtain a resource. It resists TransientError generated by
+// this call.
+// It also uses this resource's Frob function, and repairs FrobErrors
+// generated by it by calling its Defrob function.
+func Use(o ResourceOpener, inp string) (returnedError error) {
```

Comment:
When I see `o` I feel like it's a bit too short. This is more of a feeling than an objective evaluation, however. The general rule is - the shorter the lifetime, the shorter the name could be (here the lifetime is short, so maybe one letter is acceptable).
As for `inp`, I would prefer either `in` or `input`, but then again, not a thing I feel very strongly about.

Reply:
Never really thought about lifetime vs variable name size, but that sounds reasonable.